#### Variaves estatisticas



In [4]:
import pandas as pd
import numpy as np

In [6]:
train = pd.read_csv('https://raw.githubusercontent.com/nybbles/kaggle/master/train.csv')
train

survived  pclass                                               name  \
0           0       3                            Braund, Mr. Owen Harris   
1           1       1  Cumings, Mrs. John Bradley (Florence Briggs Th...   
2           1       3                             Heikkinen, Miss. Laina   
3           1       1       Futrelle, Mrs. Jacques Heath (Lily May Peel)   
4           0       3                           Allen, Mr. William Henry   
5           0       3                                   Moran, Mr. James   
6           0       1                            McCarthy, Mr. Timothy J   
7           0       3                     Palsson, Master. Gosta Leonard   
8           1       3  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)   
9           1       2                Nasser, Mrs. Nicholas (Adele Achem)   
10          1       3                    Sandstrom, Miss. Marguerite Rut   
11          1       1                           Bonnell, Miss. Elizabeth   
12          0       3                     Saundercock, Mr. William Henry   
13          0       3                        Andersson, Mr. Anders Johan   
14          0       3               Vestrom, Miss. Hulda Amanda Adolfina   
15          1       2                   Hewlett, Mrs. (Mary D Kingcome)    
16          0       3                               Rice, Master. Eugene   
17          1       2                       Williams, Mr. Charles Eugene   
18          0       3  Vander Planke, Mrs. Julius (Emelia Maria Vande...   
19          1       3                            Masselmani, Mrs. Fatima   
20          0       2                               Fynney, Mr. Joseph J   
21          1       2                              Beesley, Mr. Lawrence   
22          1       3                        McGowan, Miss. Anna "Annie"   
23          1       1                       Sloper, Mr. William Thompson   
24          0       3                      Palsson, Miss. Torborg Danira   
25          1       3  Asplund, Mrs. Carl Oscar (Selma Augusta Emilia...   
26          0       3                            Emir, Mr. Farred Chehab   
27          0       1                     Fortune, Mr. Charles Alexander   
28          1       3                      O'Dwyer, Miss. Ellen "Nellie"   
29          0       3                                Todoroff, Mr. Lalio   
..        ...     ...                                                ...   
861         0       2                        Giles, Mr. Frederick Edward   
862         1       1  Swift, Mrs. Frederick Joel (Margaret Welles Ba...   
863         0       3                  Sage, Miss. Dorothy Edith "Dolly"   
864         0       2                             Gill, Mr. John William   
865         1       2                           Bystrom, Mrs. (Karolina)   
866         1       2                       Duran y More, Miss. Asuncion   
867         0       1               Roebling, Mr. Washington Augustus II   
868         0       3                        van Melkebeke, Mr. Philemon   
869         1       3                    Johnson, Master. Harold Theodor   
870         0       3                                  Balkic, Mr. Cerin   
871         1       1   Beckwith, Mrs. Richard Leonard (Sallie Monypeny)   
872         0       1                           Carlsson, Mr. Frans Olof   
873         0       3                        Vander Cruyssen, Mr. Victor   
874         1       2              Abelson, Mrs. Samuel (Hannah Wizosky)   
875         1       3                   Najib, Miss. Adele Kiamie "Jane"   
876         0       3                      Gustafsson, Mr. Alfred Ossian   
877         0       3                               Petroff, Mr. Nedelio   
878         0       3                                 Laleff, Mr. Kristo   
879         1       1      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)   
880         1       2       Shelley, Mrs. William (Imanita Parrish Hall)   
881         0       3                                

In [64]:
f = {}
for x in train.columns:
    f[x] = pd.DataFrame(train[x].value_counts(ascending=True)

SyntaxError: unexpected EOF while parsing (<ipython-input-64-9a01b49fb5df>, line 3)

In [65]:
f={
    
}
f['age'] = train['age'].value_counts().values
train['age'].value_counts()

24.00    30
22.00    27
18.00    26
19.00    25
30.00    25
28.00    25
21.00    24
25.00    23
36.00    22
29.00    20
32.00    18
27.00    18
35.00    18
26.00    18
16.00    17
31.00    17
20.00    15
33.00    15
23.00    15
34.00    15
39.00    14
17.00    13
42.00    13
40.00    13
45.00    12
38.00    11
50.00    10
2.00     10
4.00     10
47.00     9
         ..
71.00     2
59.00     2
63.00     2
0.83      2
30.50     2
70.00     2
57.00     2
0.75      2
13.00     2
10.00     2
64.00     2
40.50     2
32.50     2
45.50     2
20.50     1
24.50     1
0.67      1
14.50     1
0.92      1
74.00     1
34.50     1
80.00     1
12.00     1
36.50     1
53.00     1
55.50     1
70.50     1
66.00     1
23.50     1
0.42      1
Name: age, Length: 88, dtype: int64

In [54]:
d = {
    'Name':['Alisa','Bobby','Cathrine','Alisa','Bobby','Cathrine',
            'Alisa','Bobby','Cathrine','Alisa','Bobby','Cathrine'],
    'Exam':['Semester 1','Semester 1','Semester 1','Semester 1','Semester 1','Semester 1',
            'Semester 2','Semester 2','Semester 2','Semester 2','Semester 2','Semester 2'],
    'Subject':['Mathematics','Mathematics','Mathematics','Science','Science','Science',
               'Mathematics','Mathematics','Mathematics','Science','Science','Science'],
    'Result':['Pass','Pass','Fail','Pass','Fail','Pass','Pass','Fail','Fail','Pass','Pass','Fail']}
 
df = pd.DataFrame(d,columns=['Name','Exam','Subject','Result'])
df

Name        Exam      Subject Result
0      Alisa  Semester 1  Mathematics   Pass
1      Bobby  Semester 1  Mathematics   Pass
2   Cathrine  Semester 1  Mathematics   Fail
3      Alisa  Semester 1      Science   Pass
4      Bobby  Semester 1      Science   Fail
5   Cathrine  Semester 1      Science   Pass
6      Alisa  Semester 2  Mathematics   Pass
7      Bobby  Semester 2  Mathematics   Fail
8   Cathrine  Semester 2  Mathematics   Fail
9      Alisa  Semester 2      Science   Pass
10     Bobby  Semester 2      Science   Pass
11  Cathrine  Semester 2      Science   Fail

In [62]:
pd.crosstab([df['Name'], df['Subject']], df['Result'])

Result                Fail  Pass
Name     Subject                
Alisa    Mathematics     0     2
         Science         0     2
Bobby    Mathematics     1     1
         Science         1     1
Cathrine Mathematics     2     0
         Science         1     1

#### Atividade: train.csv relacionar colunas fazendo perguntas

In [7]:
train

survived  pclass                                               name  \
0           0       3                            Braund, Mr. Owen Harris   
1           1       1  Cumings, Mrs. John Bradley (Florence Briggs Th...   
2           1       3                             Heikkinen, Miss. Laina   
3           1       1       Futrelle, Mrs. Jacques Heath (Lily May Peel)   
4           0       3                           Allen, Mr. William Henry   
5           0       3                                   Moran, Mr. James   
6           0       1                            McCarthy, Mr. Timothy J   
7           0       3                     Palsson, Master. Gosta Leonard   
8           1       3  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)   
9           1       2                Nasser, Mrs. Nicholas (Adele Achem)   
10          1       3                    Sandstrom, Miss. Marguerite Rut   
11          1       1                           Bonnell, Miss. Elizabeth   
12          0       3                     Saundercock, Mr. William Henry   
13          0       3                        Andersson, Mr. Anders Johan   
14          0       3               Vestrom, Miss. Hulda Amanda Adolfina   
15          1       2                   Hewlett, Mrs. (Mary D Kingcome)    
16          0       3                               Rice, Master. Eugene   
17          1       2                       Williams, Mr. Charles Eugene   
18          0       3  Vander Planke, Mrs. Julius (Emelia Maria Vande...   
19          1       3                            Masselmani, Mrs. Fatima   
20          0       2                               Fynney, Mr. Joseph J   
21          1       2                              Beesley, Mr. Lawrence   
22          1       3                        McGowan, Miss. Anna "Annie"   
23          1       1                       Sloper, Mr. William Thompson   
24          0       3                      Palsson, Miss. Torborg Danira   
25          1       3  Asplund, Mrs. Carl Oscar (Selma Augusta Emilia...   
26          0       3                            Emir, Mr. Farred Chehab   
27          0       1                     Fortune, Mr. Charles Alexander   
28          1       3                      O'Dwyer, Miss. Ellen "Nellie"   
29          0       3                                Todoroff, Mr. Lalio   
..        ...     ...                                                ...   
861         0       2                        Giles, Mr. Frederick Edward   
862         1       1  Swift, Mrs. Frederick Joel (Margaret Welles Ba...   
863         0       3                  Sage, Miss. Dorothy Edith "Dolly"   
864         0       2                             Gill, Mr. John William   
865         1       2                           Bystrom, Mrs. (Karolina)   
866         1       2                       Duran y More, Miss. Asuncion   
867         0       1               Roebling, Mr. Washington Augustus II   
868         0       3                        van Melkebeke, Mr. Philemon   
869         1       3                    Johnson, Master. Harold Theodor   
870         0       3                                  Balkic, Mr. Cerin   
871         1       1   Beckwith, Mrs. Richard Leonard (Sallie Monypeny)   
872         0       1                           Carlsson, Mr. Frans Olof   
873         0       3                        Vander Cruyssen, Mr. Victor   
874         1       2              Abelson, Mrs. Samuel (Hannah Wizosky)   
875         1       3                   Najib, Miss. Adele Kiamie "Jane"   
876         0       3                      Gustafsson, Mr. Alfred Ossian   
877         0       3                               Petroff, Mr. Nedelio   
878         0       3                                 Laleff, Mr. Kristo   
879         1       1      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)   
880         1       2       Shelley, Mrs. William (Imanita Parrish Hall)   
881         0       3                                

### 1- Quantos homens e mulheres sairam vivos?

In [70]:
pd.crosstab(train['survived'], train['sex'], margins=True)

sex       female  male  All
survived                   
0             81   468  549
1            233   109  342
All          314   577  891

** Total de pessoal a bordo:  891 **

    -Mulheres a bordo : 314
    -Homens a bordo   : 577
** Sobreviventes: 342 **

    -Mulheres 233
    -Homens 109

In [83]:
pd.crosstab([train['survived'], train['pclass']], train['sex'], margins=True)

sex              female  male  All
survived pclass                   
0        1            3    77   80
         2            6    91   97
         3           72   300  372
1        1           91    45  136
         2           70    17   87
         3           72    47  119
All                 314   577  891

### 2- Qual foi a classe que morreu mais gente e qual a que mais teve sobrevivente?

In [85]:
pd.crosstab(train['survived'], train['pclass'], margins=True)

pclass      1    2    3  All
survived                    
0          80   97  372  549
1         136   87  119  342
All       216  184  491  891

** Responsta: **

    -Classe com maior mortes: 3
    -Classe com menor mortes: 1
    -Entretanto a classe 3 tinha mais que o doblo de pessoal comparado a classe 1

### 3- Qual foi a clase que teve a maior e qual foi a que teve a menor porcentagem de mortes?

In [10]:
menor, maior = 100, 0
imenor, imaior = 0, 0
for x in range(1, 4):
    clas = pd.crosstab(train['survived'], train['pclass'], margins=True)[x]
    mortos = (clas[0] / clas[2]) * 100
    
    if menor > mortos:
        menor, imenor = mortos, x
    if maior < mortos:
        maior, imaior = mortos, x 

print(maior, menor)
print(imaior, imenor)

75.76374745417516 37.03703703703704
3 1


** Resposta: **

    -A classe 1 teve a menor porcentagem de mortes com um total de aproximadamente 75,76 %
    -A classe 3 teve a maior porcentagem de mortes com um total de aproximadamente 37,03 %